client/src/components/v2/charts/LocationTimeChart.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  AttributionControl,
+  CircleMarker,
+  Map,
+  Polyline,
+  ScaleControl,
+  TileLayer,
+} from 'react-leaflet';
+import 'leaflet/dist/leaflet.css';
+import LeafletCenterControl from 'components/LeafletCenterControl';
+import {
+  useSensorData,
+  useMultiSensorTimeSeries,
+} from 'api/v2/sensors';
+import styles from './LocationTimeChart.module.css';
+
+const MHP_WORKSHOP_LOCATION = [-37.908756, 145.13404];
+
+/**
+ * Checks if a given location is valid
+ *
+ * @param {number[]} location Location to check
+ * @returns {?Array.<number>} LatLng array if location is valid. Null if invalid
+ */
+function validateLocation(location) {
+  if (location[0] && location[1]) {
+    return location;
+  }
+  return null;
+}
+
+/**
+ * @typedef {object} LocationTimeChartProps
+ * @property {number} interval Time between updates in ms
+ */
+
+/**
+ * Location chart component
+ *
+ * @param {LocationTimeChartProps} props Props
+ * @returns {React.Component} Component
+ */
+export default function LocationTimeChart({ interval }) {
+  const { series: locationHistory } = useMultiSensorTimeSeries(['gps_lat', 'gps_long'], interval);
+  const latData = useSensorData('gps_lat');
+  const longData = useSensorData('gps_long');
+
+  const initialLocation = validateLocation([
+    latData.initialData,
+    longData.initialData,
+  ]);
+  const currentLocation = validateLocation([
+    latData.data,
+    longData.data,
+  ]);
+
+  const center = initialLocation ?? MHP_WORKSHOP_LOCATION;
```

Comment:
What does `??` do? Never seen this syntax before.

Reply:
This is the new [null coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator). It goes hand in hand with optional chaining